From ab422cf552d8705ac721890f619499a22572a7b1 Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Thu, 20 Oct 2011 10:10:37 -0700
Subject: [PATCH] Disable t9020

---
 tests/t9020-alignment.sh |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/tests/t9020-alignment.sh b/tests/t9020-alignment.sh
index 83a5c15..96491a8 100755
--- a/tests/t9020-alignment.sh
+++ b/tests/t9020-alignment.sh
@@ -34,6 +34,8 @@ modprobe -n scsi_debug ||
 grep '^#define USE_BLKID 1' "$CONFIG_HEADER" > /dev/null ||
   skip_test_ 'this system lacks a new-enough libblkid'
 
+skip_test_ "Test disabled - scsi_debug num_parts is broken on s390x and ppc64"
+
 cat <<EOF > exp || framework_failure
 minimum: 7 8
 optimal: 7 2048
-- 
1.7.6.4

