From d10470e32d132cf7a07ae8894e2d88c1bd9dbf24 Mon Sep 17 00:00:00 2001
From: Brian C. Lane <bcl@redhat.com>
Date: Thu, 7 Apr 2011 14:12:38 -0700
Subject: [PATCH] Disable tests that fail when run as root

---
 tests/t3000-resize-fs.sh |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/tests/t3000-resize-fs.sh b/tests/t3000-resize-fs.sh
index b0492db..290425f 100755
--- a/tests/t3000-resize-fs.sh
+++ b/tests/t3000-resize-fs.sh
@@ -28,6 +28,8 @@ require_root_
 require_scsi_debug_module_
 require_512_byte_sector_size_
 
+skip_test_ 'test disabled'
+
 cat <<EOF > exp-warning || framework_failure
 WARNING: you are attempting to use parted to operate on (resize) a file system.
 parted's file system manipulation code is not as robust as what you'll find in
-- 
1.7.4

