From fb2c4b49ce80c9fd4c731c2b9b972f0fed50d375 Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Wed, 19 Oct 2011 17:16:02 -0700
Subject: [PATCH] t9030 - wait for partition to be removed

---
 tests/t9030-align-check.sh |   12 ++++++++++--
 1 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/tests/t9030-align-check.sh b/tests/t9030-align-check.sh
index 12c6c77..b10d558 100644
--- a/tests/t9030-align-check.sh
+++ b/tests/t9030-align-check.sh
@@ -35,7 +35,7 @@ grep '^#define USE_BLKID 1' "$CONFIG_HEADER" > /dev/null ||
   skip_test_ 'this system lacks a new-enough libblkid'
 
 # create memory-backed device
-scsi_debug_setup_ dev_size_mb=550 physblk_exp=3 lowest_aligned=7 > dev-name ||
+scsi_debug_setup_ physblk_exp=3 lowest_aligned=7 > dev-name ||
   skip_test_ 'failed to create scsi_debug device'
 scsi_dev=$(cat dev-name)
 p1=${scsi_dev}1
@@ -46,7 +46,7 @@ parted -s $scsi_dev mklabel gpt || fail=1
 
 i=60
 while :; do
-  parted -s $scsi_dev mkpart p1 ext2 ${i}s 80000s || fail=1
+  parted -s $scsi_dev mkpart p1 ext2 ${i}s 8000s || fail=1
   wait_for_dev_to_appear_ $p1 || fail=1
   parted -s $scsi_dev align-check min 1 > out 2>&1
   result=$?
@@ -58,6 +58,14 @@ while :; do
   parted -s $scsi_dev rm 1
   i=$(expr $i + 1)
   test $i = 70 && break
+
+  # Wait until partition is gone
+  t=10
+  while [ -e $p1 ]; do
+    sleep 1
+    t=$(expr $t - 1)
+    test $t = 0 && break
+  done
 done
 
 Exit $fail
-- 
1.7.6.4

